src/mmvc/base/TriggerMap.hx

```diff
@@ -0,0 +1,141 @@
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR 
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE 
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, 
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE 
+SOFTWARE.
+*/
+
+package mmvc.base;
+
+import haxe.ds.EnumValueMap;
+import haxe.ds.ObjectMap;
+import haxe.ds.StringMap;
+
+import minject.Injector;
+
+import mmvc.api.ICommand;
+import mmvc.api.ITriggerMap;
+
+class TriggerMap implements ITriggerMap
+{
+	var injector:Injector;
+	var classToCommand:StringMap<Class<ICommand>>;
+	var stringToCommand:StringMap<Class<ICommand>>;
+	var enumValueToCommand:EnumValueMap<EnumValue, Class<ICommand>>;
+	var objectToCommand:ObjectMap<Dynamic, Class<ICommand>>;
+
+	public function new(injector:Injector)
+	{
+		this.injector = injector;
+		classToCommand = new StringMap<Class<ICommand>>();
+		stringToCommand = new StringMap<Class<ICommand>>();
+		enumValueToCommand = new EnumValueMap<EnumValue, Class<ICommand>>();
+		objectToCommand = new ObjectMap<Dynamic, Class<ICommand>>();
+	}
+
+	public function map(trigger:Class<Dynamic>, command:Class<ICommand>)
+	{
+		var triggerClassName = Type.getClassName(trigger);
+		if(classToCommand.exists(triggerClassName))
+			throw "Command for class " + triggerClassName + " is already mapped.";
+
+		classToCommand.set(triggerClassName, command);
+	}
+
+	public function unmap(trigger:Class<Dynamic>)
+	{
+		classToCommand.remove(Type.getClassName(trigger));
+	}
+
+	public function dispatch(trigger:{})
+	{
+		var triggerClass = Type.getClass(trigger);
+		var commandClass = classToCommand.get(Type.getClassName(triggerClass));
+		invokeCommand(trigger, triggerClass, commandClass);
+	}
+
+	public function mapString(trigger:String, command:Class<ICommand>)
+	{
+		if(stringToCommand.exists(trigger))
+			throw "Command for string " + trigger + " is already mapped.";
+
+		stringToCommand.set(trigger, command);
+	}
+
+	public function unmapString(trigger:String)
+	{
+		stringToCommand.remove(trigger);
+	}
+
+	public function dispatchString(trigger:String)
+	{
+		var commandClass = stringToCommand.get(trigger);
+		invokeCommand(trigger, String, commandClass);
+	}
+
+	public function mapEnumValue(trigger:EnumValue, command:Class<ICommand>)
+	{
+		if(enumValueToCommand.exists(trigger))
+			throw "Command for enum value " + Std.string(trigger) + " is already mapped.";
+
+		enumValueToCommand.set(trigger, command);
+	}
+
+	public function unmapEnumValue(trigger:EnumValue)
+	{
+		enumValueToCommand.remove(trigger);
+	}
+
+	public function dispatchEnumValue(trigger:EnumValue)
+	{
+		var triggerClass = Type.getClass(trigger);
```

Comment:
This will throw an error I think... I guess we need to be able to map EnumValues in minject (there's an issue for that already, I'll take a look at it)